enhancements/installer/openstack-customer-provided-subnets.md

```diff
@@ -0,0 +1,156 @@
+
+
+To allow the customer to set up their own custom external access infrastructure, we will have to let them set the `apiVIP` and `ingressVIP` port IPs ahead of the install, so that they can be added to their routing/loadbalancing scheme before the installer is run.
+
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: test-cluster
+platform:
+  openstack:
+    apiVIP: "192.168.30.15"
+    ingressVIP: "192.168.30.17"
+pullSecret: '{"auths": ...}'
+sshKey: ssh-ed25519 AAAA...
+```
+
+### Nodes Subnet
+- Nodes Subnet must all be a part of the same OpenStack cloud
+- The subnet must be able to be tagged, and have the capacity for at least one tag. This allows us to add the `kubernetes.io/cluster/.*:` shared tag to identify the subnet as part of the cluster. The k8s cloud provider code in kube-controller-manager uses this tag to find the subnets for the cluster to create LoadBalancer Services.
+- The CIDR block must be in MachineNetworks.
+- The host running the installer needs to be able to send HTTP requests to the API.
+- The installer should be able to support subnets on provider networks, as well as tenant networks.
+
+### Resources Created by the Installer
+When the installer is passed a set of subnets, it will no longer create a network, a subnet on that network, the ports on that subnet, the routing, or the floating ips on those networks.  
+
+The installer will continue to create:
+- Images
+- Security Groups
+- Security Group Rules
+- Ports
+- Volumes
+- Nodes
+- Boot Metadata
+
+### Installs that need more control
+For the use case where cluster operators do not want the installer creating any resources on their networks, we want them to use the UPI installer, since this usage pattern does not fit in to the IPI vision of the installer. To support this use case, we will provide additional UPI documentation.
+
+### Destroying the Cluster
+We are operating under the assumption that the users that provisioned their custom subnets will also want to manually deprovision them. Therefore, no resources will be deleted in the custom network’s or subnets that are provided to the installer. We will remove any tags we added though.
+
+All other resources will be destroyed normally.
+
+### Limitations
+This does add a bit of complexity to the install process, so it is important that we have good validation and clear documentation to help users navigate this feature.
+
+## Risks and Mitigations
```

Comment:
Do we have an environment to test and does the CI network config support this?

Reply:
I think PSI has some provider networks we can test on